util/encoding/encoding_test.go

```diff
@@ -264,3 +266,53 @@ func TestEncodeDecodeVarUint64Decreasing(t *testing.T) {
 	}
 	testCustomEncode64(testCases, EncodeVarUint64Decreasing, t)
 }
+
+func TestEncodeDecodeBytes(t *testing.T) {
+	testCases := []struct {
+		value   []byte
+		encoded []byte
+	}{
+		{[]byte{'a'}, []byte{'a', 0x00, 0x01}},
+		{[]byte{'b'}, []byte{'b', 0x00, 0x01}},
+		{[]byte{'b', 0}, []byte{'b', 0x00, 0xff, 0x00, 0x01}},
+		{[]byte{'b', 0xff}, []byte{'b', 0xff, 0x00, 0x01}},
+		{[]byte("hello"), []byte{'h', 'e', 'l', 'l', 'o', 0x00, 0x01}},
+	}
+	for i, c := range testCases {
+		enc := EncodeBytes(nil, c.value)
+		if !bytes.Equal(enc, c.encoded) {
+			t.Errorf("unexpected encoding mismatch for %v. expected %v, got %v",
+				c.value, prettyBytes(c.encoded), prettyBytes(enc))
+		}
+		if i > 0 {
+			if bytes.Compare(testCases[i-1].encoded, enc) >= 0 {
+				t.Errorf("%v: expected %v to be less than %v",
+					c.value, prettyBytes(testCases[i-1].encoded), prettyBytes(enc))
+			}
+		}
+		remainder, dec := DecodeBytes(enc)
```

Comment:
Added a test which appends `"remainder"` to the encoded value, decodes and verifies the remainder equals `"remainder"`.